Web Serial API - consistency updates (#44153)
mdn/content: files/en-us/web/api/web_serial_api/index.md

## Concepts and Usage

The Web Serial API is one of a set of APIs that allow websites to communicate with peripherals connected to a user's computer.

> It provides the ability to connect to devices that communicate via a serial port, rather than USB devices accessible via the [WebUSB API](/en-US/docs/Web/API/WebUSB_API), or input devices accessible via the [WebHID API](/en-US/docs/Web/API/WebHID_API).

Reviewer: I don't think this is quite right, although it's tricky to explain. Some devices that communicate over USB or Bluetooth will register as serial devices with the OS, and these are the devices that can use the Web Serial API. Unfortunately you just have to know when you plug in a device via USB whether it's a serial device or not.

I'm not sure the best way to communicate this, or whether it's worth it. I just want to be sure that people understand that some devices that use USB or Bluetooth can use Web Serial (as opposed to ones that connect directly via an RS-232 serial port, which are very rare these days)

Author: Yes, see https://github.com/mdn/content/pull/44153/changes#r3262864862 - can you please check it.

> Examples of serial devices include 3D printers, and microcontrollers such as the [BBC micro:bit board](https://microbit.org/).

Reviewer 2: ```suggestion
Examples of serial devices include 3D printers, ESP32 devices, and microcontrollers such as the [BBC micro:bit board](https://microbit.org/).
```

Reviewer: It may be worth mentioning ESP32 devices under the microcontrollers section here; I get the impression these are relatively popular.

## Concepts and Usage

The Web Serial API is one of a set of APIs that allow websites to communicate with peripherals connected to a user's computer.

> It provides the ability to connect to devices that communicate via a serial port, rather than USB devices accessible via the [WebUSB API](/en-US/docs/Web/API/WebUSB_API), or input devices accessible via the [WebHID API](/en-US/docs/Web/API/WebHID_API).

Author: @gregstoll You're absolutely right.

How about this:

```suggestion
The Web Serial API provides the ability to connect to devices that communicate via a serial protocol.
This includes USB and Bluetooth devices that connect over USB or Bluetooth but expose a virtual serial port to the operating system (via USB CDC-ACM or Bluetooth SPP).

Note that these are distinct from devices accessed via the  [WebUSB API](/en-US/docs/Web/API/WebUSB_API), — which provides raw access to USB devices that have not been claimed by an OS driver — or input devices that use the USB HID class, which are accessible via the [WebHID API](/en-US/docs/Web/API/WebHID_API).
```

Reviewer: Instead of "communicate via a serial port", could we say "communicate via a serial protocol"? To me a "serial port" sounds like an RS-232 port (like [these](https://readytogocables.com/blog/rs232-serial-port-7-facts-you-need-to-know-before-using/))

Thanks!

Author: Absolutely, modified above. I'll consider this done then :-) Thanks!!

The `connect` and `disconnect` events let sites react when a device is connected or disconnected from the system. The {{domxref("Serial.getPorts()","getPorts()")}} method is then called to see connected ports that the site already has access to.

If the site doesn't have access to any connected ports it has to wait until it has user activation to proceed.

> In this example we use a {{domxref("Element.click_event", "click")}} event handler on a button for this task.

Reviewer 2: ```suggestion
In this example, we use a {{domxref("Element.click_event", "click")}} event handler on a button for this task.
```
